server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -183,9 +200,9 @@ private ClusterState buildResult(ClusterState oldState, RoutingAllocation alloca
         if (restoreInProgress != null) {
             RestoreInProgress updatedRestoreInProgress = allocation.updateRestoreInfoWithRoutingChanges(restoreInProgress);
             if (updatedRestoreInProgress != restoreInProgress) {
-                ImmutableOpenMap.Builder<String, ClusterState.Custom> customsBuilder = ImmutableOpenMap.builder(allocation.getCustoms());
+                final Map<String, ClusterState.Custom> customsBuilder = new HashMap<>(allocation.getCustoms());
```

Comment:
is there a reason we made it mutable from immutable map

Reply:
Thanks for pointing out this diff. I am not sure how this diff got way. Will restore to whats there in main.